### DISCUSSION TITLE
Fetch All Threads

### DISCUSSION THREAD

**Discussion by Vidalnt** [discordjs/discord.js#9885] [Q&A] [answered]
Is it possible to get all the existing threads of a channel? I've tried that but it doesn't really get all of them, only a small half of the total number of threads.

            const channelxd = client.channels.cache.get("1101148158819573790");
            const threads = channelxd.threads.cache.filter((x) => x.isThread());
            const dataFolder = "data";    
            
            for (const info of threads.values()) {
                try {
                    await fs.mkdir(dataFolder, { recursive: true });
                    const filePath = path.join(dataFolder, `${info.id}.json`);
            
                    if (await fileExists(filePath)) {
                        jsonCount++;
                        console.log(`Ya existe <#${info.id}>`)
                        await new Promise(resolve => setTimeout(resolve, 300));
                    } else {
                        const infon = await channelxd.threads.fetch(info.id)
                        const test = await infon.fetchStarterMessage();
                        const result = {
                            id: infon.id,
                            name: infon.name,
                            owner: infon.ownerId,
                            published: infon.createdTimestamp,
                            upload: infon.createdAt,
                            tags: infon.appliedTags,
                            content: test ? test.content : null,
                            attachments: test.attachments,
                        };
            
                        await fs.mkdir(dataFolder, { recursive: true });
                        await fs.writeFile(filePath, JSON.stringify(result, null, 2));
                        jsonCount++;
                        message.channel.send(`🪄 <#${info.id}> || ${jsonCount}`);

                        await new Promise(resolve => setTimeout(resolve, 500));
                    }
and I don't know if .fetchActive or .fetchArchived help me, because I only get 25 results.

**Comment by sdanialraza** [comment, marked as answer]
Not all threads are cached, so you should fetch at the top. Not sure what you're doing with that filter, I suggest you have a look at [GuildTextThreadManager#fetch()](https://old.discordjs.dev/#/docs/discord.js/main/class/GuildTextThreadManager?scrollTo=fetch)
